palantir-java-format/src/main/java/com/palantir/javaformat/java/Main.java

```diff
@@ -177,6 +183,16 @@ private int formatFiles(CommandLineOptions parameters, JavaFormatterOptions opti
         return allOk ? 0 : 1;
     }
 
+    private void warnIfDurationExceedsThreshold(CommandLineOptions parameters, Path path, FormatFileResult fileResult) {
+        if (parameters.warnOnExpensiveFileDurationMillis().isPresent()
+                && fileResult.duration().toMillis()
+                        > parameters.warnOnExpensiveFileDurationMillis().get()) {
+            errWriter.println(path + ": took " + fileResult.duration().toMillis() + "ms to format, "
```

Comment:
nit: put `WARN` in this message, so it shows up when people Cmd+F for warnings (which log but do not fail builds)